### PR TITLE
docs: updated syntax in markdown files

### PR DESCRIPTION
**Pull request by YashasviChaurasia**
The `Note` label is updated in several places in the docs for improved rendering.
Multiple markdown files have been updated.
Closes: https://github.com/rook/rook/issues/10458

Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>

### CODE REVIEW

File: Documentation/Contributing/development-flow.md [hunk 1]
```diff
@@ -150,7 +150,8 @@ Authoring a design document for big features has many advantages:
 * Gets agreement amongst the community before code is written that could be wasted effort in the wrong direction
 * Serves as an artifact of the architecture that is easier to read for visitors to the project than just the code by itself
 
-Note that writing code to prototype the feature while working on the design may be very useful to help flesh out the approach.
+!!! Note
+That writing code to prototype the feature while working on the design may be very useful to help flesh out the approach.
```
Comment (BlaineEXE) [inline Documentation/Contributing/development-flow.md:154]:
```suggestion
!!! Note
    Writing code to prototype the feature while working on the design may be very useful to help flesh out the approach.
```

File: design/ceph/ceph-config-updates.md [hunk 2]
```diff
@@ -54,7 +54,8 @@ Notable non-required flags:
   the `/var/lib/ceph/<daemon_type>/ceph-<daemon_id>` directory for daemon data within containers. If
   possible, mapping the `dataDirHostPath/<rook_daemon_data_dir>` path on hosts to this default
   location in the container is preferred.
-  - Note that currently, `dataDirHostPath` is mapped directly to containers, meaning that each
+  !!! Note
+    Currently, `dataDirHostPath` is mapped directly to containers, meaning that each
```
Comment (BlaineEXE) [inline design/ceph/ceph-config-updates.md:58]: This isn't really a callout. Let's revert this.

```suggestion
  - Note that currently, `dataDirHostPath` is mapped directly to containers, meaning that each
```

File: design/ceph/ceph-config-updates.md [hunk 3]
```diff
@@ -135,7 +136,8 @@ spec:
       debug_osd: 10
    # ...
 ```
-**Note on the above:** all values under config are reported to Ceph as strings, but the yaml should
+!!! Note
+All values under config are reported to Ceph as strings, but the yaml should
```
Comment (BlaineEXE) [inline design/ceph/ceph-config-updates.md:140]: To render properly, callouts need to have their text indented 4 spaces (to the same level as the type of callout).

```suggestion
!!! Note
    All values under config are reported to Ceph as strings, but the yaml should
```

File: Documentation/Contributing/development-flow.md [hunk 4]
```diff
@@ -287,7 +288,8 @@ Signed-off-by: First Name Last Name <email address>
 
 The `component` **MUST** be in the [list checked by the CI](https://github.com/rook/rook/blob/master/.commitlintrc.json).
 
-Note: sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
+!!! Note
+Sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```
Comment (BlaineEXE) [inline Documentation/Contributing/development-flow.md:292]:
```suggestion
!!! Note
    Sometimes you will feel like there is not so much to say, for instance if you are fixing a typo in a text.
```

File: design/ceph/interacting-with-rook-resources.md [hunk 5]
```diff
@@ -38,6 +38,8 @@ Another example on CephFilesystem with ID=a:
...
 	app.kubernetes.io/created-by : "rook-ceph-operator"
 	rook.io/operator-namespace   : "rook-ceph"
-``` 
+```
 
-**NOTE** : A totally unique string for an application can be built up from (a) app.kubernetes.io/component, (b) app.kubernetes.io/part-of, (c) the resource's namespace, (d) app.kubernetes.io/name, and (e) app.kubernetes.io/instance fields. For the example above, we could join those fields with underscore connectors like this: cephclusters.ceph.rook.io_rook-ceph_rook-ceph_ceph-mon_a. Note that this full spec can easily exceed the 64-character limit imposed on Kubernetes labels.
\ No newline at end of file
+!!! Note : A totally unique string for an application can be built up from (a) app.kubernetes.io/component, (b) app.kubernetes.io/part-of, (c) the resource's namespace, (d) app.kubernetes.io/name, and (e) app.kubernetes.io/instance fields. For the example above, we could join those fields with underscore connectors like this: cephclusters.ceph.rook.io_rook-ceph_rook-ceph_ceph-mon_a.
+!!! Note
+This full spec can easily exceed the 64-character limit imposed on Kubernetes labels.
```
Comment (BlaineEXE) [inline design/ceph/interacting-with-rook-resources.md:45]: The phrase "note that" does not always indicate a callout. There is only one callout here. Also, indentation.

```suggestion
!!! Note
    A totally unique string for an application can be built up from (a) app.kubernetes.io/component, (b) app.kubernetes.io/part-of, (c) the resource's namespace, (d) app.kubernetes.io/name, and (e) app.kubernetes.io/instance fields. For the example above, we could join those fields with underscore connectors like this: cephclusters.ceph.rook.io_rook-ceph_rook-ceph_ceph-mon_a. Note that this full spec can easily exceed the 64-character limit imposed on Kubernetes labels.
```

**Review by BlaineEXE — CHANGES_REQUESTED** [review]
Not all instances of "note that" indicate a callout. Contextually, a standalone bullet, a block quote, or a bold bit of text like "note"/"info"/"warning"/"important" will be the best indicator.

Also, keep in mind that in order to render correctly, the text of a callout has to be indented to the same level as the type of callout (4 additional spaces).

**Comment by BlaineEXE** [comment]
Also, please make the commit message more clear. "updated syntax" is not clear. Also include the text from the PR description in the body of the commit message.

**Comment by BlaineEXE** [comment]
oops. accidentally hit close button.

**Comment by YashasviChaurasia** [comment]
@BlaineEXE is this commit message good enough?

"docs: updated `note` syntax in markdown files
The syntax to write guiding or cautionary notes in markdown files was updated.
The inconsistencies  in the syntax were removed"

Kindly inform me if you want any changes
Thank you

**Comment by YashasviChaurasia** [comment]
> Can you squash all the commits into a single commit.

Yes sure, but could you give some feedback on the commit message if I should use it or not

**Comment by BlaineEXE** [comment]
I only see this commit message:

```
docs: updated note syntax in markdown files and squashed commits
Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>
```

A few notes:

1. commit messages should only reflect changes to code. Squashing comments is a regular git process that doesn't need commented on. Let's remove "and squashed commits". Other than that, the subject line is good
2. the commit message should have a body. I think it may have gotten lost in the squashing process. I believe `git commit --amend` should allow you to quickly edit the commit message.
3. @parth-gr I generally see "resolves" and "closes" used interchangeably. That's how Github treats them at least. I think either option is fine for Rook.

**Comment by parth-gr** [comment]
```
[docs: updated note syntax in markdown files](https://github.com/rook/rook/pull/11171/commits/451d53201fdbc48db60ded49f67846e20c2c6b49) 

The Note label is updated in several places in the docs for improved rendering.
Multiple markdown files have been updated.
Closes: https://github.com/rook/rook/issues/10458

Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>
```
THere should be gap in betwween
```
[docs: updated note syntax in markdown files](https://github.com/rook/rook/pull/11171/commits/451d53201fdbc48db60ded49f67846e20c2c6b49) 

The Note label is updated in several places in the docs for improved rendering.
Multiple markdown files have been updated.
<-gap here->
Closes: https://github.com/rook/rook/issues/10458

Signed-off-by: Yashasvi Chaurasia <yashasvi12977@gmail.com>
```

**Review by parth-gr — APPROVED** [review]

**Review by BlaineEXE — APPROVED** [review]

**Comment by YashasviChaurasia** [comment]
This PR helped me learn a lot about squashing and amending commits.
Thank you, @parth-gr @BlaineEXE, for helping me figure out stuff and always being patient.
Thanks a lot

**Pull request merged**